subsys/net/ip/ipv4.c

```diff
@@ -150,12 +150,17 @@ enum net_verdict net_ipv4_input(struct net_pkt *pkt)
 	}
 
 	if (net_ipv4_is_addr_mcast(&hdr->src)) {
-		NET_DBG("DROP: src addr is mcast");
+		NET_DBG("DROP: src addr is %s", "mcast");
```

Comment:
Why DBG message is created with %s format specifier, can't it be as it was previously?

Reply:
> Why DBG message is created with %s format specifier, can't it be as it was previously?

This allows us to share the same string for multicast, broadcast and unspecified prints so we save some flash.

Reply:
> This allows us to share the same string for multicast, broadcast and unspecified prints so we save some flash.

That's of course cunning, but what about saving RAM in logging buffers? When someone enables debug logging, they want to spend flash on debug messages, but RAM always remains a scarce resource.